probe-rs/src/flashing/download.rs

```diff
@@ -224,11 +224,9 @@ fn extract_from_elf_inner<'data, T: FileHeader>(
     elf_header: &T,
     binary: ElfFile<'_, T>,
     elf_data: &'data [u8],
-) -> Result<usize, FileDownloadError> {
+) -> Result<(), FileDownloadError> {
```

Comment:
Could this return a `Vec` with the data? We only call this once, so I don't think we get any use of passing the vec as a parameter.